File: app/src/main/java/com/pgp/app/appcomponents/database/GolfClubDao.java
package com.pgp.app.appcomponents.database;
import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
public interface GolfClubDao {
        @Insert
        void insert(GolfClub golfClub);
    @Insert
    void insertAll(List<GolfClub> golfClubs);

    @Query("SELECT * FROM golf_clubs")
    LiveData<List<GolfClub>> getAllGolfClubsLiveData();


}

File: app/src/main/java/com/pgp/app/appcomponents/database/GolfClubDao.kt
package com.pgp.app.appcomponents.database

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query

@Dao
interface GolfClubDao {
    @Insert
    fun insert(golfClub: GolfClub)

    @Insert
    fun insertAll(golfClubs: List<GolfClub>)

    @get:Query("SELECT * FROM golf_clubs")
    val allGolfClubsLiveData: LiveData<List<GolfClub>>

    @Query("SELECT * FROM golf_clubs WHERE name = :clubName LIMIT 1")
    fun getGolfClubByName(clubName: String): GolfClub?
}





